core/vibes/soul/sections/reviews/product-review-button.tsx

```diff
@@ -0,0 +1,213 @@
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+      event.preventDefault();
+
+      startTransition(() => {
+        formAction(formData);
+      });
+    },
+  });
+
+  const ratingControl = useInputControl(fields.rating);
+
+  const isEmailDisabled = user.email !== '';
+  const isAuthorDisabled = user.name !== '';
+
+  useEffect(() => {
+    if (lastResult?.status === 'success' && successMessage) {
+      toast.success(successMessage);
+      setIsOpen(false);
+    }
+  }, [lastResult, successMessage]);
+
+  return (
+    <Modal
+      className="w-full md:min-w-[768px]"
+      isOpen={isOpen}
+      setOpen={setIsOpen}
+      title={formModalTitle}
+      trigger={
+        <Button className="mx-auto mt-8" size="small" variant="tertiary">
+          {formButtonLabel}
+        </Button>
+      }
+    >
+      <div className="flex flex-col gap-6 md:flex-row md:gap-8">
+        <div className="shrink-0 md:w-48">
+          <Stream
+            fallback={
+              <div className="animate-pulse">
+                <div className="mb-4 aspect-square w-full max-w-[200px] rounded-md bg-contrast-100 md:max-w-none" />
+                <div className="h-6 w-32 rounded-md bg-contrast-100" />
+              </div>
+            }
+            value={Streamable.all([streamableProduct, streamableImages])}
+          >
+            {([product, images]) => {
+              const firstImage = images[0];
+
+              return (
+                <>
+                  {firstImage && (
+                    <div className="relative mb-4 aspect-square w-full max-w-[200px] overflow-hidden rounded-md md:w-full md:max-w-none">
+                      <Image
+                        alt={firstImage.alt}
+                        className="object-cover"
+                        fill
+                        sizes="(min-width: 768px) 192px, 200px"
+                        src={firstImage.src}
+                      />
+                    </div>
+                  )}
+                  <h3 className="text-lg font-semibold">{product.name}</h3>
+                </>
+              );
+            }}
+          </Stream>
+        </div>
+        <div className="min-w-0 flex-1">
+          <form {...getFormProps(form)} action={formAction} className="flex w-full flex-col gap-5">
+            <input name="productEntityId" type="hidden" value={productId} />
+            <RatingRadioGroup
+              errors={fields.rating.errors}
+              label={formRatingLabel}
+              max={5}
+              name={fields.rating.name}
+              onBlur={ratingControl.blur}
+              onFocus={ratingControl.focus}
+              // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+              onValueChange={ratingControl.change}
+              required={fields.rating.required}
+              value={typeof ratingControl.value === 'string' ? ratingControl.value : ''}
+            />
```

Comment:
Should I just make all the form input fields controlled as well? Not sure what pattern we prefer